### PR TITLE
Alerting: Get grafana-managed alert rule by UID

### PR DESCRIPTION
**Pull request by fayzal-g**
**What is this feature?**
Adds a new API `/api/ruler/grafana/api/v1/rule/{RuleUID}` that allows the fetching of a grafana-managed alert rule, given it's UID. It will return a `403` if the user is not authorized to view the rule, and a `404` if a rule with the given UID does not exist.

**Why do we need this feature?**
We do not currently have a way to fetch a grafana-managed rule by UID, so we are resorting to fetching *ALL* rules and then matching on the given UID. This is rather inefficient and also leads to a less than desirable experience in the UI, for example when trying to view a single grafana-managed rule, 50k rules might need to be loaded.

**Special notes for your reviewer:**

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

### CODE REVIEW

File: pkg/services/ngalert/api/api_ruler.go [hunk 1]
```diff
@@ -263,6 +263,46 @@ func (srv RulerSrv) RouteGetRulesConfig(c *contextmodel.ReqContext) response.Res
...
+		Version:              rule.Version,
+		UID:                  rule.UID,
+		NamespaceUID:         rule.NamespaceUID,
+		RuleGroup:            rule.RuleGroup,
+		NoDataState:          apimodels.NoDataState(rule.NoDataState),
+		ExecErrState:         apimodels.ExecutionErrorState(rule.ExecErrState),
+		Provenance:           apimodels.Provenance(provenance),
+		IsPaused:             rule.IsPaused,
+		NotificationSettings: AlertRuleNotificationSettingsFromNotificationSettings(rule.NotificationSettings),
+	}
```
Comment (stevesg) [inline pkg/services/ngalert/api/api_ruler.go:301]: I've not spent a lot of time in this code, but it looks like the other APIs here return a `GettableExtendedRuleNode`, which has some extra fields (`For`, `Annotations`, `Labels`).

I suspect we would want to do the same here for consistency. Could we re-use the `toGettableExtendedRuleNode` function?
```suggestion
	result := toGettableExtendedRuleNode(rule, provenance)
```

**Review by stevesg — COMMENTED** [review]
Nice work - just a comment about what we should return from the API.

**Review by stevesg — APPROVED** [review]
Thanks for the changes, this looks good to go.

**Review by JohnnyQQQQ — APPROVED** [review]

**Review by gillesdemey — APPROVED** [review]

**Pull request merged**